modules/View.js

```diff
@@ -368,6 +394,7 @@
      * @method activate
      */
     activate: function() {
+      this.trigger('before-activate');
```

Comment:
i wonder if it's better to trigger it right before this._activate...

really on the fence.